torchdynamo/eval_frame.py

```diff
@@ -505,7 +505,12 @@ def output(self, target, args, kwargs):
             return super().output(target, (new_result,), {})
 
     if aten_graph:
-        graph = make_fx(graph)(*graph_captured_input)
+        # Running graph with interpreter is needed for propagating the stack_trace
+        def graph_with_interpreter(*args):
+            with torch.fx.traceback.override_stack_trace():
+                return torch.fx.Interpreter(graph).run(*args)
+
+        graph = make_fx(graph_with_interpreter)(*graph_captured_input)
```

Comment:
A critical contract of .export() is that the graph is identical to @torchdynamo.optimize(), with fx enabled, it needs to be identical to taking the @torchdynamo.optimize() + make_fx. If `graph_with_interpreter` changes the graph in any way other than setting metadata, it will no longer be equivalent.

Reply:
Added a test for this.

Reply:
Just curious, any callsites that depend on this `export == @torchdynamo.optimize() + make_fx. `? In my mental model export should return equivalent to `exir.trace` as that is what it tries to replace right now.